pandas/core/reshape/reshape.py

```diff
@@ -1014,7 +1014,8 @@ def get_empty_frame(data) -> DataFrame:
         return out
 
     else:
-        dummy_mat = np.eye(number_of_cols, dtype=dtype).take(codes, axis=0)
+        # take on axis=1 + transpose to ensure ndarray layout is column-major
```

Comment:
use the order paramater here: https://numpy.org/doc/stable/reference/generated/numpy.eye.html or maybe .reshape (as .take gives a copy)

Reply:
To clarify: the `take` is needed for the actual get_dummies operation. 
But given your suggestion, I thought that specifying `order` in `eye` might speed up the `take` call, but from a quick test that doesn't seem to matter

Reply:
guess this is fine